packages/vite/types/importMeta.d.ts

```diff
@@ -36,7 +36,7 @@ interface ImportMeta {
 }
 
 interface ImportMetaEnv {
-  [key: string]: string | boolean | undefined
+  [key: string]: any
```

Comment:
How about adding this line? (in addition to this change)
```suggestion
  [key: string]: any
  [env: `VITE_${string}`]: string
```

It requires template literal types support which is implemented in [TypeScript 4.1+](https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#template-literal-types) (released 2020/11).

This type will work like below.
```ts
/// <reference types="vite/client" />

interface ImportMetaEnv {
  VITE_FOO: string
  VITE_BAR: 'aa' | 'bbb'
  VITE_AA: number // error "this should be string" here
}
```
```ts
import.meta.env.VITE_BAR // 'aaa' | 'bbb'
import.meta.env.VITE_VAL // string
```

One thing to consider is that you can't define variables that are not string types starting with VITE with `define`.

Reply:
I think we should merge this PR in its current form for 2.9. And then, @sapphi-red maybe you could create another PR with your proposal and justify the Typescript version requirement, and we target 3.0 for the change?

Reply:
Sure. I will create a PR with more details.

Reply:
Oh there is a `envPrefix` option. It is impossible to do this approach.